app/test_main.py

```diff
@@ -1 +1,54 @@
-# write your code here
+from unittest import mock
+from typing import Callable
+
+import pytest
+
+from app.main import can_access_google_page
+
+
+@pytest.mark.parametrize(
+    "mock_has_connection, mock_is_valid_url, url, expected",
+    [
+        pytest.param(
+            True,
+            True,
+            "https://github.com/Kirontiko/",
+            "Accessible",
+            id="accessible when has connection and url is valid"
+        ),
+        pytest.param(
+            False,
+            True,
+            "https://github.com/Kirontiko/",
+            "Not accessible",
+            id="not accessible when has no connection"
+        ),
+        pytest.param(
+            True,
+            False,
+            "https://gitmug.com/3140,Kirontiko/",
+            "Not accessible",
+            id="not accessible when url is not valid"
+        ),
+        pytest.param(
+            False,
+            False,
+            "https://gitmug.com/314-K,l;.irontiko/",
+            "Not accessible",
+            id="not accessible when url is not valid and no connection"
```

Comment:
why do u need to use different urls here? 
could u use only one url and pass it directly to ur func ?

Reply:
ok, will fix right now:)